test/deployframework/helpers.go

```diff
@@ -642,6 +643,11 @@ func CreateRegistryDeployment(
 		logger.Infof("Overriding the default metering-reporting-operator image with %s", reportingOperatorImage)
 		deployment.Spec.Template.Spec.InitContainers[0].Env[1].Value = reportingOperatorImage
 	}
+	// Populate the tenant label throughout the deployment manifest
+	// that we're currently build up. By default, that label's value
+	// is set to "", so we're injecting the namespacePrefix before creation.
+	deployment.Spec.Selector.MatchLabels["tenant"] = namespacePrefix
+	deployment.Spec.Template.ObjectMeta.Labels["tenant"] = namespacePrefix
```

Comment:
This is probably a good indication that we don't really need to be using a deployment to roll out a single registry Pod as we will never need to scale up additional replicas, but I think that's out of scope out of this PR anyways.